drivers/led/led_pwm.c

```diff
@@ -68,66 +70,117 @@ static int led_pwm_set_brightness(const struct device *dev,
 			(uint32_t) ((uint64_t) dt_led->period * value / LED_BRIGHTNESS_MAX));
 }
 
-static int led_pwm_init(const struct device *dev)
+static int led_pwm_request_leds(const struct device *dev, bool request)
 {
 	const struct led_pwm_config *config = dev->config;
-	int i;
+	const struct pwm_dt_spec *led;
+	struct pm_device_runtime_reference *ref;
+	const char *request_str = request ? "request" : "release";
+	int ret = 0;
+	int err;
 
-	if (!config->num_leds) {
-		LOG_ERR("%s: no LEDs found (DT child nodes missing)",
-			dev->name);
-		return -ENODEV;
-	}
+	for (size_t i = 0; i < config->num_leds; i++) {
+		led = &config->led[i];
+		ref = &config->refs[i];
 
-	for (i = 0; i < config->num_leds; i++) {
-		const struct pwm_dt_spec *led = &config->led[i];
+		LOG_DBG("PWM %p %s", led->dev, request_str);
 
-		if (!device_is_ready(led->dev)) {
-			LOG_ERR("%s: pwm device not ready", led->dev->name);
-			return -ENODEV;
+		if (request) {
+			err = pm_device_runtime_request(led->dev, ref);
+		} else {
+			err = pm_device_runtime_release(led->dev, ref);
+		}
+
+		if (err) {
+			LOG_DBG("PWM %p %s failed (err = %d)", led->dev, request_str, ret);
+			ret = -EIO;
 		}
```

Comment:
Using `ret` instead of `err`. Should be:
```c
if (err) {
	LOG_DBG("PWM %p %s failed (err = %d)", led->dev, request_str, err);
	ret = -EIO;
}
```